### PR TITLE
chore(taxonomies): Unknown CIQUAL codes in taxonomie Yaourts

### PR DESCRIPTION
**Pull request by bernardlemoullec**
Small changes related to https://fr.openfoodfacts.org/ciqual-food-name/unknown/categorie/yaourts/categories

### CODE REVIEW

File: taxonomies/categories.txt [hunk 1]
```diff
@@ -32296,9 +32296,18 @@ nl:Drinkyoghurts
 
 <en:Drinkable yogurts
 fr:Yaourts à boire goût fraise, Yaourts à boire fraise, Yaourts à boire saveur fraise
+en:Dairy drink with sugar and fruits, Fermented milk with sugar and fruits, Yogurt with sugar and fruits
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32299]: You can add ciqual properties to multiple categories, but then you should not add the en: and fr: entries of the other category.

File: taxonomies/categories.txt [hunk 2]
```diff
@@ -32296,9 +32296,18 @@ nl:Drinkyoghurts
...
+fr:Boisson lactée sucrée au lait fermenté et aux fruits, Yaourt à boire sucré aux fruits, Lait fermenté sucré aux fruits
+ciqual_food_code:en:19535
+ciqual_food_name:en:Dairy drink or fermented milk or yogurt, with fruits, with sugar
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32302]: One issue with yogurts and drinkable yogurts is that we did not make the distinction between fruits and fruits flavors as CIQUAL did.
So "Yaourts à boire goût fraise" currently can contain both. I'm not sure what to do. cc @moon-rabbitOFF

File: taxonomies/categories.txt [hunk 3]
```diff
@@ -32341,7 +32350,11 @@ fr:Yaourts au Bifidus
 nl:Bifidus yoghurts
 
 <en:Bifidus yogurts
-fr:Yaourts au bifidus nature
+fr:Yaourts au bifidus nature, Lait fermenté de type yaourt nature au bifidus, spécialité laitière de type yaourt nature au bifidus
+en:Sweetened plain bifidus yogurts, Yogurt type plain fermented milk with sugar with bifidus, Yogurt type plain dairy specialty with sugar with bifidus
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32354]: We always put the en: entry first, to make it the canonical name of the category.

File: taxonomies/categories.txt (hunk 1 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32299]: OK, thanks for the heads up, that's exactly what I was wondering... I will fix it in my next commit then.

File: taxonomies/categories.txt (hunk 3 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32354]: Roger that, I will fix it in the next commit

File: taxonomies/categories.txt [hunk 4]
```diff
@@ -30379,7 +30379,7 @@ nl:Geraspte kzen
 
 <en:French cheeses
 fr:Maroilles, Marolles
-en:Maroilles 
+en:Maroilles
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:30382]: en: should be first

Comment (bernardlemoullec) [reply taxonomies/categories.txt:30382]: Thanks for the heads up, I will fix in the next commit

**Comment by bernardlemoullec** [comment]
I just committed some changes (b07d26b).
That's all I can do on https://fr.openfoodfacts.org/ciqual-food-name/unknown/categorie/yaourts/categories for now. I'm going to switch on to another branch.

File: taxonomies/categories.txt [hunk 5]
```diff
@@ -32726,7 +32747,7 @@ ciqual_food_name:en:Yogurt, goat's milk, plain, around 5% fat
 en:Stirred yogurts
 de:cremiger Joghurt
 fi:sekoitetut jugurtit
-fr:Yaourts brassés, Yaourt brassé
+fr:Yaourts brassés, Yaourt brassé, brasse
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32750]: "brasse" ? Unaccented values are added automatically, there is no need to add them.

Comment (bernardlemoullec) [reply taxonomies/categories.txt:32750]: Actually, I found it here: https://fr.openfoodfacts.org/ciqual-food-name/unknown/categorie/yaourts/categories

Comment (bernardlemoullec) [reply taxonomies/categories.txt:32750]: image

**Comment by teolemon** [comment]
Errors in the categories taxonomy definition:
ERROR - fr:yaourts-mixes already is a synonym of yaourts-aux-fruits (en:mixed-yogurts) - cannot make a synonym of yaourts-mixes for en:mixed-yogurts

File: taxonomies/categories.txt [hunk 6]
```diff
@@ -32292,7 +32288,7 @@ ciqual_food_name:fr:Yaourt, lait fermenté ou spécialité laitière, aux fruits
 
 <en:Yogurts
 en:Plain fermented milk yogurt with sugar, Plain dairy specialty with sugar
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32290]:
```suggestion
<en:Sweetened yogurts
<en:Plain yogurts
en:Plain sweetened yogurts, Sweetened plain yogurts, Plain fermented milk yogurt with sugar, Plain dairy specialty with sugar
```

File: taxonomies/categories.txt [hunk 7]
```diff
@@ -32292,7 +32288,7 @@ ciqual_food_name:fr:Yaourt, lait fermenté ou spécialité laitière, aux fruits
...
-fr:Yaourt sucré au lait fermenté, Spécialité laitière nature sucrée
+fr:Yaourt sucré au lait fermenté, Spécialité laitière nature sucrée, yaourts-nature-sucres
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32291]:
```suggestion
fr:Yaourts natures sucrés, Yaourts sucrés au lait fermenté, Spécialités laitières natures sucrées
```

File: taxonomies/categories.txt [hunk 8]
```diff
@@ -32331,14 +32327,30 @@ nl:Drinkyoghurts
 
 <en:Drinkable yogurts
 fr:Yaourts à boire goût fraise, Yaourts à boire fraise, Yaourts à boire saveur fraise
+en:Dairy drink with sugar and fruits, Fermented milk with sugar and fruits, Yogurt with sugar and fruits
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32330]: All categories that have <en:Drinkable yogurts as a parent need to be drinkable yogurts.
So here the first name should be
en:Sweetened drinkable yogurts with fruits
fr:Yaourts à boire sucrés aux fruits
And they should not include "Yogurt with sugar and fruits".

Could you take a look at all entries under "Drinkable yogurts"?

File: taxonomies/categories.txt [hunk 9]
```diff
@@ -32376,7 +32388,11 @@ fr:Yaourts au Bifidus
 nl:Bifidus yoghurts
 
 <en:Bifidus yogurts
-fr:Yaourts au bifidus nature
+en:Sweetened plain bifidus yogurts, Yogurt type plain fermented milk with sugar with bifidus, Yogurt type plain dairy specialty with sugar with bifidus
+fr:Yaourts au bifidus nature, Lait fermenté de type yaourt nature au bifidus, spécialité laitière de type yaourt nature au bifidus
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32392]: the fr: entry does not match the en: entry -> en is sweetened, fr is not

File: taxonomies/categories.txt [hunk 10]
```diff
@@ -32832,7 +32864,7 @@ ciqual_food_name:fr:Yaourt, lait fermenté ou spécialité laitière, nature
 
 <en:Yogurts
 en:Plain fermented milk yogurt with cream, Plain dairy specialty with cream
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32866]: When Ciqual has a category name like "ciqual_food_name:en:Yogurt, fermented milk or dairy specialty, plain, w cream"

our name should be "Plain yogurts with cream, plain fermented milks with cream, plain dairy specialties with cream".

Same for all yogurts entries. "Yaourts au lait fermenté" should just be "Yaourts, Laits fermentés".

File: taxonomies/categories.txt (hunk 6 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32290]: Fixed in the next commit

File: taxonomies/categories.txt (hunk 7 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32291]: Fixed in the next commit

File: taxonomies/categories.txt (hunk 8 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32330]: Done, thanks for the heads up.
Some CIQUAL translations are a bit ambiguous. For instance:
ciqual_food_code:en:19508
ciqual_food_name:en:Dairy drink or fermented milk or yogurt, flavoured, with sugar
ciqual_food_name:fr:Boisson lactée, lait fermenté ou yaourt à boire, aromatisé, sucré

File: taxonomies/categories.txt (hunk 9 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32392]: I think that one was a bad copy-paste, thanks for the info :)

File: taxonomies/categories.txt (hunk 10 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32866]: Fixed in the next commit

File: taxonomies/categories.txt [hunk 11]
```diff
@@ -32815,39 +32812,53 @@ hu:Ivójoghurtok, Iható joghurtok
 nl:Drinkyoghurts
 
 <en:Drinkable yogurts
-fr:Yaourts à boire goût fraise, Yaourts à boire fraise, Yaourts à boire saveur fraise
+fr:Yaourts à boire goût vanille, Yaourts à boire vanille, Yaourts à boire saveur vanille
+ciqual_food_code:en:19508
 
 <en:Drinkable yogurts
-fr:Yaourts à boire goût vanille, Yaourts à boire vanille, Yaourts à boire saveur vanille
+fr:Yaourts à boire goût framboise, yaourts-a-boire-gout-framboise
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32819]: All the strings like yaourts-a-boire-gout-framboise need to be removed if there are already existing corresponding synonyms that are correctly accented / capitalized.

File: taxonomies/categories.txt [hunk 12]
```diff
@@ -32805,7 +32802,7 @@ ciqual_food_name:fr:Yaourt, lait fermenté ou spécialité laitière, aromatisé
 
 <en:Yogurts
 <en:Fermented milk drinks
-en:Drinkable yogurts
+en:Drinkable yogurts, yogurt-drink
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:32805]: we canonicalize "yogurt drink" to yogurt-drink, but in the synonyms, we should list the correct name : " yogurt drink" (with accents, without the dashes)

File: taxonomies/categories.txt (hunk 11 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32819]: OK, done for yaourts-a-boire

File: taxonomies/categories.txt (hunk 12 above)
Comment (bernardlemoullec) [reply taxonomies/categories.txt:32805]: OK, although I'm not sure I understand what you mean by "canonicalize" in this context :)

**Comment by aleene** [comment]
I made an attempt to remove conflicts

**Comment by benbenben2** [comment]
Hi @teolemon  and @moon-rabbitOFF 
Last commit is an attempt to resolve conflicts for this old PR.

**Comment by benbenben2** [comment]
@stephanegigandet @aleene @aleene 
If someone can review and eventually approve this PR. So that it can be closed.

**Review by stephanegigandet — APPROVED** [review]
Thank you everyone!

**Pull request merged**